observability/src/test/java/io/grpc/observability/interceptors/GcpLogHelperTest.java

```diff
@@ -0,0 +1,565 @@
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.nio.charset.Charset;
+import java.util.concurrent.TimeUnit;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link GcpLogHelper}. */
+@RunWith(JUnit4.class)
+public class GcpLogHelperTest {
+  private static final Charset US_ASCII = Charset.forName("US-ASCII");
+  private static final String DATA_A = "aaaaaaaaa";
+  private static final String DATA_B = "bbbbbbbbb";
+  private static final String DATA_C = "ccccccccc";
+  private static final Metadata.Key<String> KEY_A =
+      Metadata.Key.of("a", Metadata.ASCII_STRING_MARSHALLER);
+  private static final Metadata.Key<String> KEY_B =
+      Metadata.Key.of("b", Metadata.ASCII_STRING_MARSHALLER);
+  private static final Metadata.Key<String> KEY_C =
+      Metadata.Key.of("c", Metadata.ASCII_STRING_MARSHALLER);
+  private static final MetadataEntry ENTRY_A =
+      MetadataEntry
+          .newBuilder()
+          .setKey(KEY_A.name())
+          .setValue(ByteString.copyFrom(DATA_A.getBytes(US_ASCII)))
+          .build();
+  private static final MetadataEntry ENTRY_B =
+      MetadataEntry
+          .newBuilder()
+          .setKey(KEY_B.name())
+          .setValue(ByteString.copyFrom(DATA_B.getBytes(US_ASCII)))
+          .build();
+  private static final MetadataEntry ENTRY_C =
+      MetadataEntry
+          .newBuilder()
+          .setKey(KEY_C.name())
+          .setValue(ByteString.copyFrom(DATA_C.getBytes(US_ASCII)))
+          .build();
+
+  private final Metadata nonEmptyMetadata = new Metadata();
+  private final int nonEmptyMetadataSize = 10;
+  private final Sink sink = mock(GcpLogSink.class);
+  private final Timestamp timestamp
+      = Timestamp.newBuilder().setSeconds(9876).setNanos(54321).build();
+  private final TimeProvider timeProvider = new TimeProvider() {
+    @Override
+    public long currentTimeNanos() {
+      return TimeUnit.SECONDS.toNanos(9876) + 54321;
+    }
+  };
+  private final GcpLogSinkWriter sinkWriter =
+      new GcpLogSinkWriter(
+          sink,
+          timeProvider);
+  private final byte[] message = new byte[100];
+  private SocketAddress peer;
+
+  @Before
+  public void setUp() throws Exception {
+    nonEmptyMetadata.put(KEY_A, DATA_A);
+    nonEmptyMetadata.put(KEY_B, DATA_B);
+    nonEmptyMetadata.put(KEY_C, DATA_C);
+    peer = new InetSocketAddress(InetAddress.getByName("127.0.0.1"), 1234);
+  }
+
+  @Test
+  public void socketToProto_ipv4() throws Exception {
+    InetAddress address = InetAddress.getByName("127.0.0.1");
+    int port = 12345;
+    InetSocketAddress socketAddress = new InetSocketAddress(address, port);
+    assertEquals(
+        Address
+            .newBuilder()
+            .setType(Address.Type.TYPE_IPV4)
+            .setAddress("127.0.0.1")
+            .setIpPort(12345)
+            .build(),
+        GcpLogHelper.socketToProto(socketAddress));
```

Comment:
Instead of comparing 2 protos may be you can just convert the value returned from `GcpLogHelper.socketToProto` to a string (bytestring?) and compare with a string constant?

Reply:
Do you mean converting both the protos to string constant and comparing them? 
I  thought by comparing fields of the proto returned from `GcpLogHelper.socketAddressToProto` we can also verify conversion of InetAddress to String. 
Let me know if I am missing something.